crypto/cloudkms/signer.go

```diff
@@ -22,113 +22,115 @@ import (
 	"context"
 	"encoding/asn1"
 	"fmt"
+	"hash/crc32"
 	"math/big"
 
 	kms "cloud.google.com/go/kms/apiv1"
-	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
-
 	"github.com/onflow/flow-go-sdk/crypto"
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+	"google.golang.org/protobuf/types/known/wrapperspb"
 )
 
+var _ crypto.Signer = (*Signer)(nil)
+
 // Signer is a Google Cloud KMS implementation of crypto.Signer.
 type Signer struct {
-	ctx      context.Context
-	client   *kms.KeyManagementClient
-	key      Key
-	hashAlgo crypto.HashAlgorithm
+	ctx    context.Context
+	client *kms.KeyManagementClient
+	key    Key
+	// ECDSA is the only algorithm supported by this package. The signature algorithm
+	// therefore represents the elliptic curve used. The curve is needed to parse the kms signature.
+	curve crypto.SignatureAlgorithm
+	// public key for easier access
+	publicKey crypto.PublicKey
 }
 
-// SignerForKey returns a new Google Cloud KMS signer for an asymmetric key version.
+// SignerForKey returns a new Google Cloud KMS signer for an asymmetric signing key version.
+//
+// Only ECDSA keys on P-256 and secp256k1 curves and SHA2-256 are supported.
 func (c *Client) SignerForKey(
 	ctx context.Context,
 	key Key,
 ) (*Signer, error) {
-	_, hashAlgo, err := c.GetPublicKey(ctx, key)
+
+	pk, _, err := c.GetPublicKey(ctx, key)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Signer{
-		ctx:      ctx,
-		client:   c.client,
-		key:      key,
-		hashAlgo: hashAlgo,
+		ctx:       ctx,
+		client:    c.client,
+		key:       key,
+		curve:     pk.Algorithm(),
+		publicKey: pk,
 	}, nil
 }
 
 // Sign signs the given message using the KMS signing key for this signer.
 //
 // Reference: https://cloud.google.com/kms/docs/create-validate-signatures
 func (s *Signer) Sign(message []byte) ([]byte, error) {
-	hasher, err := crypto.NewHasher(s.hashAlgo)
-	if err != nil {
-		return nil, fmt.Errorf("cloudkms: failed to instantiate hasher: %w", err)
-	}
-	digest := hasher.ComputeHash(message)
-
-	digestMessage, err := makeDigest(s.hashAlgo, digest)
-	if err != nil {
-		return nil, fmt.Errorf("cloudkms: failed to construct digest: %w", err)
-	}
 
 	request := &kmspb.AsymmetricSignRequest{
-		Name:   s.key.ResourceID(),
-		Digest: digestMessage,
+		Name:       s.key.ResourceID(),
```

Comment:
This change was not tested and requires testing with a KMS. How do you usually test these KMS changes ?  @psiemens @judezhu 
Ideally, the test would be added to CI (maybe have a kms key (ring) accessible by CI if that's simple) ?

Reply:
I was testing that manually before. You need to have access to KMS (contact SRE) and then create a key (there should also be one already created by me a while ago). Like you said adding this to the CI and making an automatic test for it would be way better 💯

Reply:
hey @tarakby, yeah! same here. In my local testing, we were using a `kms` key hosted on GCP. You probably can mock the `kms` sign function: `AsymmetricSign`, if you wanna get rid of the `kms` dependency in our local testing here.